### PR TITLE
Move UAX#14 defines to line.toml

### PR DESCRIPTION
**Pull request by makotokato**
- Generate UAX#14 table using toml file like UAX#29.
- Remove python tool to generate UAX#14 machine state table

### CODE REVIEW

File: experimental/segmenter/build.rs [hunk 1]
```diff
@@ -86,7 +89,9 @@ fn set_break_state(
     right_index: usize,
     break_state: i8,
 ) {
-    if break_state_table[left_index * property_length + right_index] == UNKNOWN_RULE {
+    if break_state_table[left_index * property_length + right_index] == UNKNOWN_RULE
+        || break_state_table[left_index * property_length + right_index] == NOT_MATCH_RULE
+    {
         break_state_table[left_index * property_length + right_index] = break_state;
     }
 }
```
Comment (aethanyc) [inline experimental/segmenter/build.rs:97]: Nit: Maybe use a variable to store `left_index * property_length + right_index`?

**Review by aethanyc — COMMENTED** [review]
I'm excited to see python script gets converted to TOML! I assume this doesn't change the behavior?

Suggestion:
1. Remove `rule_table.rs` since it's longer used.
2. How about renaming `line_breaker.rs` to `line.rs` to match other breaker implementation like `word.rs`, etc?

**Comment by makotokato** [comment]
And I also add multi thread version of data generation. But I will improve more after this.

**Review by aethanyc — APPROVED** [review]
> And I also add multi thread version of data generation. But I will improve more after this.

It's nice to speed up `bulid.rs` as the interim developer efficiency. However, I'd feel we probably shouldn't spend more time on the performance of `bulid.rs` because we really should migrate the segmenter data generation into icu4x data generation tool, i.e. https://github.com/unicode-org/icu4x/tree/main/tools/datagen.  Ideally, the segmenter data generation can be speed up on top of the work in #1600.

**Review by aethanyc — APPROVED** [review]

**Pull request merged**